### PR TITLE
Wait for wallet address to be set before naving to NewAddress screen

### LINKED ISSUE #394: Wallet address doesn't show up until neutrino has begun syncing
Related to https://github.com/lightningnetwork/lnd/issues/444

We could work around this issue by displaying the wait screen (spinner) until `store.walletAddress` is set.

### PR DESCRIPTION
**Pull request by valentinewallace**
Closes #394 
Now we wait for the wallet address to be non-null before navigating to the NewAddress screen

### CODE REVIEW

File: src/action/wallet.js [hunk 1]
```diff
@@ -261,6 +261,25 @@ class WalletAction {
...
 
+  /**
+   * Ensure that the wallet address is non-null before navigating to the
+   * NewAddress view during onboarding.
+   * This is necessary because the wallet address may be null if neutrino
+   * has not started syncing by the time the user finishes verifying
+   * their seed.
+   * @return {undefined}
+   */
+  addInitialCoin() {
```
Comment (tanx) [inline src/action/wallet.js]: Rename to `initInitialDeposit` for consistency.

File: src/action/wallet.js [hunk 2]
```diff
@@ -261,6 +261,25 @@ class WalletAction {
...
+    if (this._store.walletAddress !== null) {
```
Comment (tanx) [inline src/action/wallet.js]: Checking if `this._store.walletAddress` is truthy should be cleaner.

File: src/action/wallet.js [hunk 3]
```diff
@@ -261,6 +261,25 @@ class WalletAction {
...
+      this._nav.goNewAddress();
+    } else {
+      this._nav.goWait();
+      observe(this._store, 'walletAddress', () => {
+        this._nav.goNewAddress();
+      });
```
Comment (tanx) [inline src/action/wallet.js]: Remove `{ }` and inline.

File: src/action/wallet.js [hunk 4]
```diff
@@ -261,6 +261,25 @@ class WalletAction {
...
+    }
+  }
+
```
Comment (tanx) [inline src/action/wallet.js]: Unit tests? 🙃

File: src/action/wallet.js (hunk 1 above)
Comment (valentinewallace) [reply src/action/wallet.js]: fixed

File: src/action/wallet.js (hunk 2 above)
Comment (valentinewallace) [reply src/action/wallet.js]: fixed

File: src/action/wallet.js (hunk 3 above)
Comment (valentinewallace) [reply src/action/wallet.js]: fixed

File: src/action/wallet.js (hunk 4 above)
Comment (valentinewallace) [reply src/action/wallet.js]: whooooops fixed :)

**Comment by erkarl** [comment]
utACK

File: src/view/seed-success.js [hunk 5]
```diff
@@ -40,13 +40,15 @@ const SeedSuccessView = ({ nav }) => (
           }
         </CopyText>
       </View>
-      <GlasButton onPress={() => nav.goNewAddress()}>Add some coin</GlasButton>
+      <GlasButton onPress={() => wallet.addInitialCoin()}>
```
Comment (tanx) [inline src/view/seed-success.js]: Update to renamed `initInitialDeposit`.

File: test/unit/action/wallet.spec.js [hunk 6]
```diff
@@ -221,6 +221,23 @@ describe('Action Wallet Unit Tests', () => {
...
+    it('should navigate to new address screen if address is non-null', () => {
+      store.walletAddress = 'non-null-addr';
+      wallet.initInitialDeposit();
+      expect(nav.goNewAddress, 'was called once');
+      expect(nav.goWait, 'was not called');
+    });
+
+    it('should stay on wait screen until address is non-null', async () => {
+      store.walletAddress = null;
+      wallet.initInitialDeposit();
```
Comment (tanx) [inline test/unit/action/wallet.spec.js]: Add check that `expect(nav.goNewAddress, 'was not called');` was not called before setting the store value.

File: src/view/seed-success.js (hunk 5 above)
Comment (erkarl) [reply src/view/seed-success.js]: Nice catch :).

Comment (valentinewallace) [reply src/view/seed-success.js]: Jeez, must've been tired. fixed, thanks :)

File: test/unit/action/wallet.spec.js (hunk 6 above)
Comment (valentinewallace) [reply test/unit/action/wallet.spec.js]: fixed

**Pull request merged**